### PR TITLE
Introduce backup handler for upgrade

### PR DESCRIPTION
**Pull request by alf-cactus**
link #1032 
@tpurschke if this is fine with you I will use handlers for the other roles as well

### CODE REVIEW

File: documentation/operations.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Aug 25 16:48:19 ubu2004test fworch.middleware-server[403234]: FWO_Auth_Server::L
 ==> ldap.log <==
 Aug 25 16:48:19 ubu2004test slapd[403175]: conn=1002 fd=12 ACCEPT from IP=127.0.0.1:35558 (IP=0.0.0.0:636)
 Aug 25 16:48:19 ubu2004test slapd[403175]: conn=1002 fd=12 TLS established tls_ssf=256 ssf=256
-Aug 25 16:48:19 ubu2004test slapd[403175]: conn=1002 op=0 BIND dn="uid=admin,ou=systemuser,ou=user,dc=fworch,dc=internal" method=128
+Aug 25 16:48:19 ubu2004test slapd[403175]: conn=1002 op=0 BIND dn="uid=admin,ou=systemuser,ou=user,h,dc=internal" method=128
```
Comment (tpurschke) [inline documentation/operations.md:24]: probably not intentionally?

**Review by tpurschke — CHANGES_REQUESTED** [review]
upgrade test (with failing ms dotnet download running on ubuntu 18.04) failed with missing ui dir:
```
TASK [middleware : set_fact] *********************************************************************************************************************************************************************************
skipping: [fworch-srv]

TASK [middleware : get package list from ms packages-microsoft-prod.deb] *************************************************************************************************************************************
fatal: [fworch-srv]: FAILED! => {"changed": false, "dest": "/usr/local/fworch/middleware/files/packages-microsoft-prod.deb", "elapsed": 10, "msg": "Request failed: <urlopen error [Errno -3] Temporary failure in name resolution>", "url": "https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb"}

RUNNING HANDLER [middleware handler] *************************************************************************************************************************************************************************
changed: [fworch-srv]

PLAY RECAP ***************************************************************************************************************************************************************************************************
fworch-srv                 : ok=95   changed=27   unreachable=0    failed=1    skipped=111  rescued=0    ignored=0   


tim@ubu18test:~/firewall-orchestrator$ ls -l /usr/local/fworch/
total 28
drwxr-xr-x  5 fworch fworch 4096 Jul 19 18:33 database
drwxr-xr-x  3 fworch fworch 4096 Jul 19 18:10 etc
drwxr-xr-x 17 root   root   4096 Jul 19 18:32 fworch_backup
drwxr-xr-x  3 fworch fworch 4096 Jul 19 18:33 lib
drwxr-xr-x  3 fworch fworch 4096 Jul 19 18:34 middleware
drwxr-xr-x  3 root   root   4096 Jul 19 18:33 openldap
drwxr-xr-x  3 fworch fworch 4096 Jul 19 18:11 test
tim@ubu18test:~/firewall-orchestrator$ 
```

File: documentation/operations.md (hunk 1 above)
Comment (alf-cactus) [reply documentation/operations.md]: Thank you for your vigilance

**Review by tpurschke — CHANGES_REQUESTED** [review]
error with current status on ubuntu 18.04:
```
tim@ubu18test:~/firewall-orchestrator$ ansible-playbook site.yml -e "testkeys=yes installation_mode=uninstall" -K 
BECOME password: 
ERROR! conflicting action statements: ansible.posix.synchronize, delegate

The error appears to be in '/home/tim/firewall-orchestrator/roles/middleware/tasks/main.yml': line 11, column 5, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


  - name: backup middleware dir
    ^ here
tim@ubu18test:~/firewall-orchestrator$ 
```

**Comment by alf-cactus** [comment]
The only thing missing is the guarantee that all dirs are present. I get errors when i.e. ui dir is missing

**Review by tpurschke — CHANGES_REQUESTED** [review]
looking much better
after a mw server restart, the failed upgrade is working
can you add a (mw) server restart?

**Review by tpurschke — CHANGES_REQUESTED** [review]
suggestion: in case of failed upgrade, could we print out a warning at the very end, e.g.
middleware server could not be upgraded (due to connectivity issues with Microsoft), please try to run the upgrade later.

**Comment by alf-cactus** [comment]
Works as intended for ubuntu 20.04

**Review by tpurschke — APPROVED** [review]
test with ubuntu 18.04 was also successful

**Pull request merged**